// ui-library/dist/es/Button-CqJ10Q2z.js
import { defineComponent as C, computed as _, openBlock as B, createElementBlock as X, mergeProps as at, createVNode as R, unref as x, normalizeProps as Mt, guardReactiveProps as Vt, ref as he, watchEffect as Nt, watch as Gt, provide as me, renderSlot as O, inject as _e, normalizeClass as Q, createElementVNode as K, createTextVNode as Rt, toDisplayString as Wt, Transition as Ht, toHandlers as Xt, withCtx as ge, withDirectives as qt, vShow as Yt, useSlots as it, createBlock as ee, resolveDynamicComponent as ut, normalizeStyle as te, createCommentVNode as ne, reactive as ct, toRef as D } from "vue";
import { FontAwesomeIcon as Jt } from "@fortawesome/vue-fontawesome";
import { library as Kt } from "@fortawesome/fontawesome-svg-core";
import { fas as Zt } from "@fortawesome/free-solid-svg-icons";
import { t as Qt } from "./vendor-BVNR0mO6.js";
import { _ as lt, w as st } from "./utils-DSsG9kky.js";
var ft = typeof global == "object" && global && global.Object === Object && global, en = typeof self == "object" && self && self.Object === Object && self, g = ft || en || Function("return this")(), j = g.Symbol, pt = Object.prototype, tn = pt.hasOwnProperty, nn = pt.toString, N = j ? j.toStringTag : void 0;
function rn(e) {
  var t = tn.call(e, N), n = e[N];
  try {
    e[N] = void 0;
    var r = !0;
  } catch {
  }
  var o = nn.call(e);
  return r && (t ? e[N] = n : delete e[N]), o;
}
var on = Object.prototype, an = on.toString;
function un(e) {
  return an.call(e);
}
var cn = "[object Null]", ln = "[object Undefined]", xe = j ? j.toStringTag : void 0;
function E(e) {
  return e == null ? e === void 0 ? ln : cn : xe && xe in Object(e) ? rn(e) : un(e);
}
function k(e) {
  return e != null && typeof e == "object";
}
var sn = "[object Symbol]";
function ae(e) {
  return typeof e == "symbol" || k(e) && E(e) == sn;
}
function dt(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length, o = Array(r); ++n < r; )
    o[n] = t(e[n], n, e);
  return o;
}
var P = Array.isArray, fn = 1 / 0, Se = j ? j.prototype : void 0, Ie = Se ? Se.toString : void 0;
function vt(e) {
  if (typeof e == "string")
    return e;
  if (P(e))
    return dt(e, vt) + "";
  if (ae(e))
    return Ie ? Ie.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -fn ? "-0" : t;
}
var pn = /\s/;
function dn(e) {
  for (var t = e.length; t-- && pn.test(e.charAt(t)); )
    ;
  return t;
}
var vn = /^\s+/;
function yn(e) {
  return e && e.slice(0, dn(e) + 1).replace(vn, "");
}
function S(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var Te = NaN, bn = /^[-+]0x[0-9a-f]+$/i, hn = /^0b[01]+$/i, mn = /^0o[0-7]+$/i, _n = parseInt;
function Ce(e) {
  if (typeof e == "number")
    return e;
  if (ae(e))
    return Te;
  if (S(e)) {
    var t = typeof e.valueOf == "function" ? e.valueOf() : e;
    e = S(t) ? t + "" : t;
  }
  if (typeof e != "string")
    return e === 0 ? e : +e;
  e = yn(e);
  var n = hn.test(e);
  return n || mn.test(e) ? _n(e.slice(2), n ? 2 : 8) : bn.test(e) ? Te : +e;
}
function gn(e) {
  return e;
}
var jn = "[object AsyncFunction]", wn = "[object Function]", Bn = "[object GeneratorFunction]", On = "[object Proxy]";
function yt(e) {
  if (!S(e))
    return !1;
  var t = E(e);
  return t == wn || t == Bn || t == jn || t == On;
}
var le = g["__core-js_shared__"], Ee = function() {
  var e = /[^.]+$/.exec(le && le.keys && le.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function $n(e) {
  return !!Ee && Ee in e;
}
var An = Function.prototype, zn = An.toString;
function F(e) {
  if (e != null) {
    try {
      return zn.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var xn = /[\\^$.*+?()[\]{}|]/g, Sn = /^\[object .+?Constructor\]$/, In = Function.prototype, Tn = Object.prototype, Cn = In.toString, En = Tn.hasOwnProperty, kn = RegExp(
  "^" + Cn.call(En).replace(xn, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Pn(e) {
  if (!S(e) || $n(e))
    return !1;
  var t = yt(e) ? kn : Sn;
  return t.test(F(e));
}
function Fn(e, t) {
  return e == null ? void 0 : e[t];
}
function U(e, t) {
  var n = Fn(e, t);
  return Pn(n) ? n : void 0;
}
var de = U(g, "WeakMap");
function Un(e, t, n) {
  switch (n.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, n[0]);
    case 2:
      return e.call(t, n[0], n[1]);
    case 3:
      return e.call(t, n[0], n[1], n[2]);
  }
  return e.apply(t, n);
}
var Dn = 800, Ln = 16, Mn = Date.now;
function Vn(e) {
  var t = 0, n = 0;
  return function() {
    var r = Mn(), o = Ln - (r - n);
    if (n = r, o > 0) {
      if (++t >= Dn)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function Nn(e) {
  return function() {
    return e;
  };
}
var re = function() {
  try {
    var e = U(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}(), Gn = re ? function(e, t) {
  return re(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: Nn(t),
    writable: !0
  });
} : gn, Rn = Vn(Gn);
function Wn(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length; ++n < r && t(e[n], n, e) !== !1; )
    ;
  return e;
}
var Hn = 9007199254740991, Xn = /^(?:0|[1-9]\d*)$/;
function qn(e, t) {
  var n = typeof e;
  return t = t ?? Hn, !!t && (n == "number" || n != "symbol" && Xn.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function bt(e, t, n) {
  t == "__proto__" && re ? re(e, t, {
    configurable: !0,
    enumerable: !0,
    value: n,
    writable: !0
  }) : e[t] = n;
}
function ht(e, t) {
  return e === t || e !== e && t !== t;
}
var Yn = Object.prototype, Jn = Yn.hasOwnProperty;
function mt(e, t, n) {
  var r = e[t];
  (!(Jn.call(e, t) && ht(r, n)) || n === void 0 && !(t in e)) && bt(e, t, n);
}
function Kn(e, t, n, r) {
  var o = !n;
  n || (n = {});
  for (var a = -1, c = t.length; ++a < c; ) {
    var i = t[a], l = void 0;
    l === void 0 && (l = e[i]), o ? bt(n, i, l) : mt(n, i, l);
  }
  return n;
}
var ke = Math.max;
function Zn(e, t, n) {
  return t = ke(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var r = arguments, o = -1, a = ke(r.length - t, 0), c = Array(a); ++o < a; )
      c[o] = r[t + o];
    o = -1;
    for (var i = Array(t + 1); ++o < t; )
      i[o] = r[o];
    return i[t] = n(c), Un(e, this, i);
  };
}
var Qn = 9007199254740991;
function _t(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Qn;
}
function er(e) {
  return e != null && _t(e.length) && !yt(e);
}
var tr = Object.prototype;
function nr(e) {
  var t = e && e.constructor, n = typeof t == "function" && t.prototype || tr;
  return e === n;
}
function rr(e, t) {
  for (var n = -1, r = Array(e); ++n < e; )
    r[n] = t(n);
  return r;
}
var or = "[object Arguments]";
function Pe(e) {
  return k(e) && E(e) == or;
}
var gt = Object.prototype, ar = gt.hasOwnProperty, ir = gt.propertyIsEnumerable, jt = Pe(/* @__PURE__ */ function() {
  return arguments;
}()) ? Pe : function(e) {
  return k(e) && ar.call(e, "callee") && !ir.call(e, "callee");
};
function ur() {
  return !1;
}
var wt = typeof exports == "object" && exports && !exports.nodeType && exports, Fe = wt && typeof module == "object" && module && !module.nodeType && module, cr = Fe && Fe.exports === wt, Ue = cr ? g.Buffer : void 0, lr = Ue ? Ue.isBuffer : void 0, Bt = lr || ur, sr = "[object Arguments]", fr = "[object Array]", pr = "[object Boolean]", dr = "[object Date]", vr = "[object Error]", yr = "[object Function]", br = "[object Map]", hr = "[object Number]", mr = "[object Object]", _r = "[object RegExp]", gr = "[object Set]", jr = "[object String]", wr = "[object WeakMap]", Br = "[object ArrayBuffer]", Or = "[object DataView]", $r = "[object Float32Array]", Ar = "[object Float64Array]", zr = "[object Int8Array]", xr = "[object Int16Array]", Sr = "[object Int32Array]", Ir = "[object Uint8Array]", Tr = "[object Uint8ClampedArray]", Cr = "[object Uint16Array]", Er = "[object Uint32Array]", p = {};
p[$r] = p[Ar] = p[zr] = p[xr] = p[Sr] = p[Ir] = p[Tr] = p[Cr] = p[Er] = !0;
p[sr] = p[fr] = p[Br] = p[pr] = p[Or] = p[dr] = p[vr] = p[yr] = p[br] = p[hr] = p[mr] = p[_r] = p[gr] = p[jr] = p[wr] = !1;
function kr(e) {
  return k(e) && _t(e.length) && !!p[E(e)];
}
function je(e) {
  return function(t) {
    return e(t);
  };
}
var Ot = typeof exports == "object" && exports && !exports.nodeType && exports, G = Ot && typeof module == "object" && module && !module.nodeType && module, Pr = G && G.exports === Ot, se = Pr && ft.process, L = function() {
  try {
    var e = G && G.require && G.require("util").types;
    return e || se && se.binding && se.binding("util");
  } catch {
  }
}(), De = L && L.isTypedArray, Fr = De ? je(De) : kr;
function Ur(e, t) {
  var n = P(e), r = !n && jt(e), o = !n && !r && Bt(e), a = !n && !r && !o && Fr(e), c = n || r || o || a, i = c ? rr(e.length, String) : [], l = i.length;
  for (var s in e)
    !(c && // Safari 9 has enumerable `arguments.length` in strict mode.
    (s == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    o && (s == "offset" || s == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    a && (s == "buffer" || s == "byteLength" || s == "byteOffset") || // Skip index properties.
    qn(s, l))) && i.push(s);
  return i;
}
function Dr(e, t) {
  return function(n) {
    return e(t(n));
  };
}
function Lr(e) {
  var t = [];
  if (e != null)
    for (var n in Object(e))
      t.push(n);
  return t;
}
var Mr = Object.prototype, Vr = Mr.hasOwnProperty;
function Nr(e) {
  if (!S(e))
    return Lr(e);
  var t = nr(e), n = [];
  for (var r in e)
    r == "constructor" && (t || !Vr.call(e, r)) || n.push(r);
  return n;
}
function Gr(e) {
  return er(e) ? Ur(e) : Nr(e);
}
var Rr = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Wr = /^\w*$/;
function Hr(e, t) {
  if (P(e))
    return !1;
  var n = typeof e;
  return n == "number" || n == "symbol" || n == "boolean" || e == null || ae(e) ? !0 : Wr.test(e) || !Rr.test(e) || t != null && e in Object(t);
}
var W = U(Object, "create");
function Xr() {
  this.__data__ = W ? W(null) : {}, this.size = 0;
}
function qr(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Yr = "__lodash_hash_undefined__", Jr = Object.prototype, Kr = Jr.hasOwnProperty;
function Zr(e) {
  var t = this.__data__;
  if (W) {
    var n = t[e];
    return n === Yr ? void 0 : n;
  }
  return Kr.call(t, e) ? t[e] : void 0;
}
var Qr = Object.prototype, eo = Qr.hasOwnProperty;
function to(e) {
  var t = this.__data__;
  return W ? t[e] !== void 0 : eo.call(t, e);
}
var no = "__lodash_hash_undefined__";
function ro(e, t) {
  var n = this.__data__;
  return this.size += this.has(e) ? 0 : 1, n[e] = W && t === void 0 ? no : t, this;
}
function T(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
T.prototype.clear = Xr;
T.prototype.delete = qr;
T.prototype.get = Zr;
T.prototype.has = to;
T.prototype.set = ro;
function oo() {
  this.__data__ = [], this.size = 0;
}
function ie(e, t) {
  for (var n = e.length; n--; )
    if (ht(e[n][0], t))
      return n;
  return -1;
}
var ao = Array.prototype, io = ao.splice;
function uo(e) {
  var t = this.__data__, n = ie(t, e);
  if (n < 0)
    return !1;
  var r = t.length - 1;
  return n == r ? t.pop() : io.call(t, n, 1), --this.size, !0;
}
function co(e) {
  var t = this.__data__, n = ie(t, e);
  return n < 0 ? void 0 : t[n][1];
}
function lo(e) {
  return ie(this.__data__, e) > -1;
}
function so(e, t) {
  var n = this.__data__, r = ie(n, e);
  return r < 0 ? (++this.size, n.push([e, t])) : n[r][1] = t, this;
}
function $(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
$.prototype.clear = oo;
$.prototype.delete = uo;
$.prototype.get = co;
$.prototype.has = lo;
$.prototype.set = so;
var H = U(g, "Map");
function fo() {
  this.size = 0, this.__data__ = {
    hash: new T(),
    map: new (H || $)(),
    string: new T()
  };
}
function po(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function ue(e, t) {
  var n = e.__data__;
  return po(t) ? n[typeof t == "string" ? "string" : "hash"] : n.map;
}
function vo(e) {
  var t = ue(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function yo(e) {
  return ue(this, e).get(e);
}
function bo(e) {
  return ue(this, e).has(e);
}
function ho(e, t) {
  var n = ue(this, e), r = n.size;
  return n.set(e, t), this.size += n.size == r ? 0 : 1, this;
}
function I(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var r = e[t];
    this.set(r[0], r[1]);
  }
}
I.prototype.clear = fo;
I.prototype.delete = vo;
I.prototype.get = yo;
I.prototype.has = bo;
I.prototype.set = ho;
var mo = "Expected a function";
function we(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(mo);
  var n = function() {
    var r = arguments, o = t ? t.apply(this, r) : r[0], a = n.cache;
    if (a.has(o))
      return a.get(o);
    var c = e.apply(this, r);
    return n.cache = a.set(o, c) || a, c;
  };
  return n.cache = new (we.Cache || I)(), n;
}
we.Cache = I;
var _o = 500;
function go(e) {
  var t = we(e, function(r) {
    return n.size === _o && n.clear(), r;
  }), n = t.cache;
  return t;
}
var jo = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, wo = /\\(\\)?/g, Bo = go(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(jo, function(n, r, o, a) {
    t.push(o ? a.replace(wo, "$1") : r || n);
  }), t;
});
function Oo(e) {
  return e == null ? "" : vt(e);
}
function Be(e, t) {
  return P(e) ? e : Hr(e, t) ? [e] : Bo(Oo(e));
}
var $o = 1 / 0;
function $t(e) {
  if (typeof e == "string" || ae(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -$o ? "-0" : t;
}
function Ao(e, t) {
  t = Be(t, e);
  for (var n = 0, r = t.length; e != null && n < r; )
    e = e[$t(t[n++])];
  return n && n == r ? e : void 0;
}
function Oe(e, t) {
  for (var n = -1, r = t.length, o = e.length; ++n < r; )
    e[o + n] = t[n];
  return e;
}
var Le = j ? j.isConcatSpreadable : void 0;
function zo(e) {
  return P(e) || jt(e) || !!(Le && e && e[Le]);
}
function xo(e, t, n, r, o) {
  var a = -1, c = e.length;
  for (n || (n = zo), o || (o = []); ++a < c; ) {
    var i = e[a];
    n(i) ? Oe(o, i) : o[o.length] = i;
  }
  return o;
}
function So(e) {
  var t = e == null ? 0 : e.length;
  return t ? xo(e) : [];
}
function Io(e) {
  return Rn(Zn(e, void 0, So), e + "");
}
var At = Dr(Object.getPrototypeOf, Object), To = "[object Object]", Co = Function.prototype, Eo = Object.prototype, zt = Co.toString, ko = Eo.hasOwnProperty, Po = zt.call(Object);
function Fo(e) {
  if (!k(e) || E(e) != To)
    return !1;
  var t = At(e);
  if (t === null)
    return !0;
  var n = ko.call(t, "constructor") && t.constructor;
  return typeof n == "function" && n instanceof n && zt.call(n) == Po;
}
function Uo(e, t, n) {
  var r = -1, o = e.length;
  t < 0 && (t = -t > o ? 0 : o + t), n = n > o ? o : n, n < 0 && (n += o), o = t > n ? 0 : n - t >>> 0, t >>>= 0;
  for (var a = Array(o); ++r < o; )
    a[r] = e[r + t];
  return a;
}
function Do() {
  this.__data__ = new $(), this.size = 0;
}
function Lo(e) {
  var t = this.__data__, n = t.delete(e);
  return this.size = t.size, n;
}
function Mo(e) {
  return this.__data__.get(e);
}
function Vo(e) {
  return this.__data__.has(e);
}
var No = 200;
function Go(e, t) {
  var n = this.__data__;
  if (n instanceof $) {
    var r = n.__data__;
    if (!H || r.length < No - 1)
      return r.push([e, t]), this.size = ++n.size, this;
    n = this.__data__ = new I(r);
  }
  return n.set(e, t), this.size = n.size, this;
}
function M(e) {
  var t = this.__data__ = new $(e);
  this.size = t.size;
}
M.prototype.clear = Do;
M.prototype.delete = Lo;
M.prototype.get = Mo;
M.prototype.has = Vo;
M.prototype.set = Go;
var xt = typeof exports == "object" && exports && !exports.nodeType && exports, Me = xt && typeof module == "object" && module && !module.nodeType && module, Ro = Me && Me.exports === xt, Ve = Ro ? g.Buffer : void 0;
Ve && Ve.allocUnsafe;
function Wo(e, t) {
  return e.slice();
}
function Ho(e, t) {
  for (var n = -1, r = e == null ? 0 : e.length, o = 0, a = []; ++n < r; ) {
    var c = e[n];
    t(c, n, e) && (a[o++] = c);
  }
  return a;
}
function St() {
  return [];
}
var Xo = Object.prototype, qo = Xo.propertyIsEnumerable, Ne = Object.getOwnPropertySymbols, Yo = Ne ? function(e) {
  return e == null ? [] : (e = Object(e), Ho(Ne(e), function(t) {
    return qo.call(e, t);
  }));
} : St, Jo = Object.getOwnPropertySymbols, Ko = Jo ? function(e) {
  for (var t = []; e; )
    Oe(t, Yo(e)), e = At(e);
  return t;
} : St;
function Zo(e, t, n) {
  var r = t(e);
  return P(e) ? r : Oe(r, n(e));
}
function It(e) {
  return Zo(e, Gr, Ko);
}
var ve = U(g, "DataView"), ye = U(g, "Promise"), be = U(g, "Set"), Ge = "[object Map]", Qo = "[object Object]", Re = "[object Promise]", We = "[object Set]", He = "[object WeakMap]", Xe = "[object DataView]", ea = F(ve), ta = F(H), na = F(ye), ra = F(be), oa = F(de), w = E;
(ve && w(new ve(new ArrayBuffer(1))) != Xe || H && w(new H()) != Ge || ye && w(ye.resolve()) != Re || be && w(new be()) != We || de && w(new de()) != He) && (w = function(e) {
  var t = E(e), n = t == Qo ? e.constructor : void 0, r = n ? F(n) : "";
  if (r)
    switch (r) {
      case ea:
        return Xe;
      case ta:
        return Ge;
      case na:
        return Re;
      case ra:
        return We;
      case oa:
        return He;
    }
  return t;
});
var aa = Object.prototype, ia = aa.hasOwnProperty;
function ua(e) {
  var t = e.length, n = new e.constructor(t);
  return t && typeof e[0] == "string" && ia.call(e, "index") && (n.index = e.index, n.input = e.input), n;
}
var qe = g.Uint8Array;
function $e(e) {
  var t = new e.constructor(e.byteLength);
  return new qe(t).set(new qe(e)), t;
}
function ca(e, t) {
  var n = $e(e.buffer);
  return new e.constructor(n, e.byteOffset, e.byteLength);
}
var la = /\w*$/;
function sa(e) {
  var t = new e.constructor(e.source, la.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var Ye = j ? j.prototype : void 0, Je = Ye ? Ye.valueOf : void 0;
function fa(e) {
  return Je ? Object(Je.call(e)) : {};
}
function pa(e, t) {
  var n = $e(e.buffer);
  return new e.constructor(n, e.byteOffset, e.length);
}
var da = "[object Boolean]", va = "[object Date]", ya = "[object Map]", ba = "[object Number]", ha = "[object RegExp]", ma = "[object Set]", _a = "[object String]", ga = "[object Symbol]", ja = "[object ArrayBuffer]", wa = "[object DataView]", Ba = "[object Float32Array]", Oa = "[object Float64Array]", $a = "[object Int8Array]", Aa = "[object Int16Array]", za = "[object Int32Array]", xa = "[object Uint8Array]", Sa = "[object Uint8ClampedArray]", Ia = "[object Uint16Array]", Ta = "[object Uint32Array]";
function Ca(e, t, n) {
  var r = e.constructor;
  switch (t) {
    case ja:
      return $e(e);
    case da:
    case va:
      return new r(+e);
    case wa:
      return ca(e);
    case Ba:
    case Oa:
    case $a:
    case Aa:
    case za:
    case xa:
    case Sa:
    case Ia:
    case Ta:
      return pa(e);
    case ya:
      return new r();
    case ba:
    case _a:
      return new r(e);
    case ha:
      return sa(e);
    case ma:
      return new r();
    case ga:
      return fa(e);
  }
}
var Ea = "[object Map]";
function ka(e) {
  return k(e) && w(e) == Ea;
}
var Ke = L && L.isMap, Pa = Ke ? je(Ke) : ka, Fa = "[object Set]";
function Ua(e) {
  return k(e) && w(e) == Fa;
}
var Ze = L && L.isSet, Da = Ze ? je(Ze) : Ua, La = 1, Tt = "[object Arguments]", Ma = "[object Array]", Va = "[object Boolean]", Na = "[object Date]", Ga = "[object Error]", Ct = "[object Function]", Ra = "[object GeneratorFunction]", Wa = "[object Map]", Ha = "[object Number]", Et = "[object Object]", Xa = "[object RegExp]", qa = "[object Set]", Ya = "[object String]", Ja = "[object Symbol]", Ka = "[object WeakMap]", Za = "[object ArrayBuffer]", Qa = "[object DataView]", ei = "[object Float32Array]", ti = "[object Float64Array]", ni = "[object Int8Array]", ri = "[object Int16Array]", oi = "[object Int32Array]", ai = "[object Uint8Array]", ii = "[object Uint8ClampedArray]", ui = "[object Uint16Array]", ci = "[object Uint32Array]", f = {};
f[Tt] = f[Ma] = f[Za] = f[Qa] = f[Va] = f[Na] = f[ei] = f[ti] = f[ni] = f[ri] = f[oi] = f[Wa] = f[Ha] = f[Et] = f[Xa] = f[qa] = f[Ya] = f[Ja] = f[ai] = f[ii] = f[ui] = f[ci] = !0;
f[Ga] = f[Ct] = f[Ka] = !1;
function Z(e, t, n, r, o, a) {
  var c, i = t & La;
  if (n && (c = o ? n(e, r, o, a) : n(e)), c !== void 0)
    return c;
  if (!S(e))
    return e;
  var l = P(e);
  if (l)
    c = ua(e);
  else {
    var s = w(e), d = s == Ct || s == Ra;
    if (Bt(e))
      return Wo(e);
    if (s == Et || s == Tt || d && !o)
      c = {};
    else {
      if (!f[s])
        return o ? e : {};
      c = Ca(e, s);
    }
  }
  a || (a = new M());
  var b = a.get(e);
  if (b)
    return b;
  a.set(e, c), Da(e) ? e.forEach(function(u) {
    c.add(Z(u, t, n, u, e, a));
  }) : Pa(e) && e.forEach(function(u, y) {
    c.set(y, Z(u, t, n, y, e, a));
  });
  var m = It, h = l ? void 0 : m(e);
  return Wn(h || e, function(u, y) {
    h && (y = u, u = e[y]), mt(c, y, Z(u, t, n, y, e, a));
  }), c;
}
var fe = function() {
  return g.Date.now();
}, li = "Expected a function", si = Math.max, fi = Math.min;
function pi(e, t, n) {
  var r, o, a, c, i, l, s = 0, d = !1, b = !1, m = !0;
  if (typeof e != "function")
    throw new TypeError(li);
  t = Ce(t) || 0, S(n) && (d = !!n.leading, b = "maxWait" in n, a = b ? si(Ce(n.maxWait) || 0, t) : a, m = "trailing" in n ? !!n.trailing : m);
  function h(v) {
    var z = r, V = o;
    return r = o = void 0, s = v, c = e.apply(V, z), c;
  }
  function u(v) {
    return s = v, i = setTimeout(J, t), d ? h(v) : c;
  }
  function y(v) {
    var z = v - l, V = v - s, ze = t - z;
    return b ? fi(ze, a - V) : ze;
  }
  function A(v) {
    var z = v - l, V = v - s;
    return l === void 0 || z >= t || z < 0 || b && V >= a;
  }
  function J() {
    var v = fe();
    if (A(v))
      return Ae(v);
    i = setTimeout(J, y(v));
  }
  function Ae(v) {
    return i = void 0, m && r ? h(v) : (r = o = void 0, c);
  }
  function Dt() {
    i !== void 0 && clearTimeout(i), s = 0, r = l = o = i = void 0;
  }
  function Lt() {
    return i === void 0 ? c : Ae(fe());
  }
  function ce() {
    var v = fe(), z = A(v);
    if (r = arguments, o = this, l = v, z) {
      if (i === void 0)
        return u(l);
      if (b)
        return clearTimeout(i), i = setTimeout(J, t), h(l);
    }
    return i === void 0 && (i = setTimeout(J, t)), c;
  }
  return ce.cancel = Dt, ce.flush = Lt, ce;
}
function di(e) {
  var t = e == null ? 0 : e.length;
  return t ? e[t - 1] : void 0;
}
function vi(e, t) {
  return t.length < 2 ? e : Ao(e, Uo(t, 0, -1));
}
function yi(e, t) {
  return t = Be(t, e), e = vi(e, t), e == null || delete e[$t(di(t))];
}
function bi(e) {
  return Fo(e) ? void 0 : e;
}
var hi = 1, mi = 2, _i = 4, gi = Io(function(e, t) {
  var n = {};
  if (e == null)
    return n;
  var r = !1;
  t = dt(t, function(a) {
    return a = Be(a, e), r || (r = a.length > 1), a;
  }), Kn(e, It(e), n), r && (n = Z(n, hi | mi | _i, bi));
  for (var o = t.length; o--; )
    yi(n, t[o]);
  return n;
}), ji = "Expected a function";
function wi(e, t, n) {
  var r = !0, o = !0;
  if (typeof e != "function")
    throw new TypeError(ji);
  return S(n) && (r = "leading" in n ? !!n.leading : r, o = "trailing" in n ? !!n.trailing : o), pi(e, t, {
    leading: r,
    maxWait: t,
    trailing: o
  });
}
const q = (e) => (e && (e.install = (t) => {
  const n = (e == null ? void 0 : e.name) || "UnnamedComponent";
  t.component(n, e);
}), e);
class Bi extends Error {
  // 调用了父类 Error 的构造函数，传入错误消息 message
  constructor(t) {
    super(t), this.name = "LiUIError";
  }
}
function Qe(e, t) {
  if (process.env.NODE_ENV !== "production") {
    const n = new Bi(`[${e}] ${t}`);
    console.warn(n);
  }
}
const Y = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [r, o] of t)
    n[r] = o;
  return n;
}, Oi = /* @__PURE__ */ C({
  name: "LiIcon",
  inheritAttrs: !1,
  __name: "Icon",
  props: {
    border: { type: Boolean },
    fixedWidth: { type: Boolean },
    flip: {},
    icon: {},
    mask: {},
    listItem: { type: Boolean },
    pull: {},
    pulse: { type: Boolean },
    rotation: {},
    swapOpacity: { type: Boolean },
    size: {},
    spin: { type: Boolean },
    transform: {},
    symbol: { type: [Boolean, String] },
    title: {},
    inverse: { type: Boolean },
    bounce: { type: Boolean },
    shake: { type: Boolean },
    beat: { type: Boolean },
    fade: { type: Boolean },
    beatFade: { type: Boolean },
    spinPulse: { type: Boolean },
    spinReverse: { type: Boolean },
    type: {},
    color: {}
  },
  setup(e) {
    const t = e, n = _(() => gi(t, ["type", "color"])), r = _(() => ({ color: t.color ?? void 0 }));
    return (o, a) => (B(), X("i", at({
      class: ["li-icon", [`li-icon-${t.type}`]],
      style: r.value
    }, o.$attrs), [
      R(x(Jt), Mt(Vt(n.value)), null, 16)
    ], 16));
  }
}), kt = /* @__PURE__ */ Y(Oi, [["__scopeId", "data-v-4ac1c3ce"]]), oe = q(kt), Pt = Symbol("collapseContext"), $i = { class: "li-collapse" }, pe = "LiCollapse", Ai = /* @__PURE__ */ C({
  name: pe,
  __name: "Collapse",
  props: {
    modelValue: {},
    accordion: { type: Boolean }
  },
  emits: ["update:modelValue", "change"],
  setup(e, { emit: t }) {
    const n = e, r = t, o = he(n.modelValue);
    n.accordion && o.value.length > 1 && Qe(pe, "accordion mode should only hace one active item");
    function a(i) {
      let l = [...o.value];
      if (n.accordion) {
        l = [l[0] === i ? "" : i], c(l);
        return;
      }
      const s = l.indexOf(i);
      s > -1 ? l.splice(s, 1) : l.push(i), c(l);
    }
    function c(i) {
      o.value = i, r("update:modelValue", i), r("change", i);
    }
    return Nt(() => {
      n.accordion && o.value.length > 1 && Qe(pe, "accordion mode should only hace one active item");
    }), Gt(
      () => n.modelValue,
      (i) => c(i)
    ), me(Pt, {
      activeNames: o,
      handleItemClick: a
    }), (i, l) => (B(), X("div", $i, [
      O(i.$slots, "default", {}, void 0, !0)
    ]));
  }
}), zi = /* @__PURE__ */ Y(Ai, [["__scopeId", "data-v-3464ba23"]]), et = (e) => e.style.height = "0px", tt = (e) => e.style.height = `${e.scrollHeight}px`, nt = (e) => e.style.height = "", rt = (e) => e.style.overflow = "hidden", ot = (e) => e.style.overflow = "", xi = {
  beforeEnter(e) {
    et(e), rt(e);
  },
  enter: (e) => tt(e),
  afterEnter(e) {
    nt(e), ot(e);
  },
  beforeLeave(e) {
    tt(e), rt(e);
  },
  leave: (e) => et(e),
  afterLeave(e) {
    nt(e), ot(e);
  }
}, Si = ["id"], Ii = { class: "li-collapse-item__title" }, Ti = { class: "li-collapse-item__wapper" }, Ci = ["id"], Ei = /* @__PURE__ */ C({
  name: "LiCollapseItem",
  __name: "CollapseItem",
  props: {
    name: {},
    title: {},
    disabled: { type: Boolean }
  },
  setup(e) {
    const t = e, n = _e(Pt, void 0), r = _(() => {
      var a;
      return (a = n == null ? void 0 : n.activeNames.value) == null ? void 0 : a.includes(t.name);
    });
    function o() {
      t.disabled || n == null || n.handleItemClick(t.name);
    }
    return (a, c) => (B(), X("div", {
      class: Q(["li-collapse-item", {
        "is-disabled": a.disabled
      }])
    }, [
      K("div", {
        class: Q(["li-collapse-item__header", {
          "is-disabled": a.disabled,
          "is-active": r.value
        }]),
        id: `item-header-${a.name}`,
        onClick: o
      }, [
        K("span", Ii, [
          O(a.$slots, "title", {}, () => [
            Rt(Wt(a.title), 1)
          ], !0)
        ]),
        R(kt, {
          icon: "angle-right",
          class: "header-angle"
        })
      ], 10, Si),
      R(Ht, at({ name: "slide" }, Xt(x(xi))), {
        default: ge(() => [
          qt(K("div", Ti, [
            K("div", {
              class: "li-collapse-item__content",
              id: `item-content-${a.name}`
            }, [
              O(a.$slots, "default", {}, void 0, !0)
            ], 8, Ci)
          ], 512), [
            [Yt, r.value]
          ])
        ]),
        _: 3
      }, 16)
    ], 2));
  }
}), ki = /* @__PURE__ */ Y(Ei, [["__scopeId", "data-v-d8f4308e"]]);
q(zi);
q(ki);
const Ft = Symbol("buttonGroupContext"), Pi = /* @__PURE__ */ C({
  name: "LiButton",
  __name: "Button",
  props: {
    tag: { default: "button" },
    type: {},
    size: {},
    nativeType: { default: "button" },
    disabled: { type: Boolean },
    loading: { type: Boolean },
    icon: {},
    circle: { type: Boolean },
    plain: { type: Boolean },
    round: { type: Boolean },
    loadingIcon: {},
    autofocus: { type: Boolean },
    useThrottle: { type: Boolean, default: !0 },
    throttleDuration: { default: 500 }
  },
  emits: ["click"],
  setup(e, { expose: t, emit: n }) {
    const r = e, o = n, a = it(), c = he(), i = _(() => ({
      marginRight: a.default ? "6px" : "0px"
    })), l = (u) => o("click", u), s = wi(
      l,
      r.throttleDuration,
      // 节流的时间间隔
      { trailing: !1 }
      // 只有在节流时间间隔的开始时调用函数
    );
    t({
      ref: c
    });
    const d = _e(Ft, void 0), b = _(() => (d == null ? void 0 : d.size) ?? (r == null ? void 0 : r.size) ?? ""), m = _(() => (d == null ? void 0 : d.type) ?? (r == null ? void 0 : r.type) ?? ""), h = _(() => (d == null ? void 0 : d.disabled) || (r == null ? void 0 : r.disabled) || !1);
    return (u, y) => (B(), ee(ut(u.tag), {
      ref_key: "_ref",
      ref: c,
      class: Q(["li-button", {
        [`li-button--${m.value}`]: m.value,
        [`li-button--${b.value}`]: b.value,
        "is-plain": u.plain,
        "is-round": u.round,
        "is-circle": u.circle,
        "is-disabled": h.value,
        "is-loading": u.loading
      }]),
      autofocus: u.autofocus,
      type: u.tag === "button" ? u.nativeType : void 0,
      disabled: h.value || u.loading ? !0 : void 0,
      onClick: y[0] || (y[0] = (A) => u.useThrottle ? x(s)(A) : l(A))
    }, {
      default: ge(() => [
        u.loading ? O(u.$slots, "loading", { key: 0 }, () => [
          R(x(oe), {
            class: "loading-icon",
            icon: u.loadingIcon ?? "spinner",
            style: te(i.value),
            size: "1x",
            spin: ""
          }, null, 8, ["icon", "style"])
        ], !0) : ne("", !0),
        u.icon && !u.loading ? (B(), ee(x(oe), {
          key: 1,
          icon: u.icon,
          style: te(i.value),
          size: "1x"
        }, null, 8, ["icon", "style"])) : ne("", !0),
        O(u.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
    }, 8, ["autofocus", "type", "disabled", "class"]));
  }
}), Fi = /* @__PURE__ */ Y(Pi, [["__scopeId", "data-v-0fd715d9"]]), Ui = { class: "li-button-group" }, Di = /* @__PURE__ */ C({
  name: "LiButtonGroup",
  __name: "ButtonGroup",
  props: {
    size: {},
    type: {},
    disabled: { type: Boolean }
  },
  setup(e) {
    const t = e;
    return me(
      Ft,
      ct({
        size: D(t, "size"),
        type: D(t, "type"),
        disabled: D(t, "disabled")
      })
    ), (n, r) => (B(), X("div", Ui, [
      O(n.$slots, "default", {}, void 0, !0)
    ]));
  }
}), Li = /* @__PURE__ */ Y(Di, [["__scopeId", "data-v-4ba70178"]]);
q(Fi);
q(Li);
Kt.add(Zt);
const Ut = Symbol("buttonGroupContext"), Mi = /* @__PURE__ */ C({
  name: "LiButton",
  __name: "Button",
  props: {
    tag: { default: "button" },
    type: {},
    size: {},
    nativeType: { default: "button" },
    disabled: { type: Boolean },
    loading: { type: Boolean },
    icon: {},
    circle: { type: Boolean },
    plain: { type: Boolean },
    round: { type: Boolean },
    loadingIcon: {},
    autofocus: { type: Boolean },
    useThrottle: { type: Boolean, default: !0 },
    throttleDuration: { default: 500 }
  },
  emits: ["click"],
  setup(e, { expose: t, emit: n }) {
    const r = e, o = n, a = it(), c = he(), i = _(() => ({
      marginRight: a.default ? "6px" : "0px"
    })), l = (u) => o("click", u), s = Qt(
      l,
      r.throttleDuration,
      // 节流的时间间隔
      { trailing: !1 }
      // 只有在节流时间间隔的开始时调用函数
    );
    t({
      ref: c
    });
    const d = _e(Ut, void 0), b = _(() => (d == null ? void 0 : d.size) ?? (r == null ? void 0 : r.size) ?? ""), m = _(() => (d == null ? void 0 : d.type) ?? (r == null ? void 0 : r.type) ?? ""), h = _(() => (d == null ? void 0 : d.disabled) || (r == null ? void 0 : r.disabled) || !1);
    return (u, y) => (B(), ee(ut(u.tag), {
      ref_key: "_ref",
      ref: c,
      class: Q(["li-button", {
        [`li-button--${m.value}`]: m.value,
        [`li-button--${b.value}`]: b.value,
        "is-plain": u.plain,
        "is-round": u.round,
        "is-circle": u.circle,
        "is-disabled": h.value,
        "is-loading": u.loading
      }]),
      autofocus: u.autofocus,
      type: u.tag === "button" ? u.nativeType : void 0,
      disabled: h.value || u.loading ? !0 : void 0,
      onClick: y[0] || (y[0] = (A) => u.useThrottle ? x(s)(A) : l(A))
    }, {
      default: ge(() => [
        u.loading ? O(u.$slots, "loading", { key: 0 }, () => [
          R(x(oe), {
            class: "loading-icon",
            icon: u.loadingIcon ?? "spinner",
            style: te(i.value),
            size: "1x",
            spin: ""
          }, null, 8, ["icon", "style"])
        ], !0) : ne("", !0),
        u.icon && !u.loading ? (B(), ee(x(oe), {
          key: 1,
          icon: u.icon,
          style: te(i.value),
          size: "1x"
        }, null, 8, ["icon", "style"])) : ne("", !0),
        O(u.$slots, "default", {}, void 0, !0)
      ]),
      _: 3
    }, 8, ["autofocus", "type", "disabled", "class"]));
  }
}), Vi = /* @__PURE__ */ lt(Mi, [["__scopeId", "data-v-0fd715d9"]]), Ni = { class: "li-button-group" }, Gi = /* @__PURE__ */ C({
  name: "LiButtonGroup",
  __name: "ButtonGroup",
  props: {
    size: {},
    type: {},
    disabled: { type: Boolean }
  },
  setup(e) {
    const t = e;
    return me(
      Ut,
      ct({
        size: D(t, "size"),
        type: D(t, "type"),
        disabled: D(t, "disabled")
      })
    ), (n, r) => (B(), X("div", Ni, [
      O(n.$slots, "default", {}, void 0, !0)
    ]));
  }
}), Ri = /* @__PURE__ */ lt(Gi, [["__scopeId", "data-v-4ba70178"]]), Ki = st(Vi), Zi = st(Ri);
export {
  Ki as L,
  Zi as a
};
