// ui-library/dist/es/index.js
import { m as _ } from "./utils-CYlLLZae.js";
import { L as o, a as n } from "./Button-B-XyGPPC.js";
import { a as t } from "./Icon-D6P40Jo1.js";
import { L as r, a as s } from "./Collapse-U4h6vw2d.js";
import { L as a } from "./Alert-BaLM4sZq.js";
import { L as e } from "./Tooltip-CMBpjfBs.js";
import { library as i } from "@fortawesome/fontawesome-svg-core";
import { fas as m } from "@fortawesome/free-solid-svg-icons";
const f = [o, n, t, r, s, a, e];
!function() {
  {
    const _2 = "\n  ______________________________________________________________\n  \n   _______ _______       _____   __ __                          \n  |   |   |_     _|_____|     |_|__|  |--.----.---.-.----.--.--.\n  |   |   |_|   ||______|       |  |  _  |   _|  _  |   _|  |  |\n  |_______|_______|     |_______|__|_____|__| |___._|__| |___  |\n                                                         |_____|                                           \n  ______________________________________________________________\n                           author: LiXiao\n  ", o2 = "\n        background: linear-gradient(135deg, orange 60%, cyan);\n        background-clip: text;\n        color: transparent;\n        font-size: 16px; \n        line-height: 1;\n        font-family: monospace;\n        font-weight: 600;\n  ";
    console.info(`%c${_2}`, o2);
  }
}(), i.add(m);
const c = _(f);
export {
  a as LiAlert,
  o as LiButton,
  n as LiButtonGroup,
  r as LiCollapse,
  s as LiCollapseItem,
  t as LiIcon,
  e as LiTooltip,
  c as default
};
