// ui-library/dist/es/index.js
import { m as _ } from "./utils-C8KhF7Wa.js";
import { L as o, a as n } from "./Button-2DxHuayI.js";
import { a as t } from "./Icon-D-sR320q.js";
import { L as a, a as r } from "./Collapse-BrSuAyYd.js";
import { L as s } from "./Alert-593Ndcap.js";
import { library as e } from "@fortawesome/fontawesome-svg-core";
import { fas as i } from "@fortawesome/free-solid-svg-icons";
const m = [o, n, t, a, r, s];
!function() {
  {
    const _2 = "\n  ______________________________________________________________\n  \n   _______ _______       _____   __ __                          \n  |   |   |_     _|_____|     |_|__|  |--.----.---.-.----.--.--.\n  |   |   |_|   ||______|       |  |  _  |   _|  _  |   _|  |  |\n  |_______|_______|     |_______|__|_____|__| |___._|__| |___  |\n                                                         |_____|                                           \n  ______________________________________________________________\n                           author: LiXiao\n  ", o2 = "\n        background: linear-gradient(135deg, orange 60%, cyan);\n        background-clip: text;\n        color: transparent;\n        font-size: 16px; \n        line-height: 1;\n        font-family: monospace;\n        font-weight: 600;\n  ";
    console.info(`%c${_2}`, o2);
  }
}(), e.add(i);
const f = _(m);
export {
  s as LiAlert,
  o as LiButton,
  n as LiButtonGroup,
  a as LiCollapse,
  r as LiCollapseItem,
  t as LiIcon,
  f as default
};
